contracts/OysterPearl.sol

```diff
@@ -172,26 +179,27 @@ contract OysterPearl {
    *
    * When an address is buried; only claimAmount can be withdrawn once per epoch
    */
-  function bury() public returns (bool success) {
-    // The address must be previously unburied
-    require(!buried[msg.sender]);
-
-    // An address must have at least claimAmount to be buried
-    require(balances[msg.sender] >= claimAmount);
-
-    // Prevent addresses with large balances from getting buried
-    require(balances[msg.sender] <= retentionMax);
-
-    // Set buried state to true
-    buried[msg.sender] = true;
-
-    // Set the initial claim clock to 1
-    claimed[msg.sender] = 1;
-
-    // Execute an event reflecting the change
-    Bury(msg.sender, balances[msg.sender]);
-    return true;
-  }
+   function bury(bytes32 hash) public returns (bool success) {
+     // The hash must be previously unburied
+     require(!buried[hash]);
+        
+     // claimAmount is able to be transferred to this contract
+     require(_transfer(msg.sender, address(this), claimAmount));
+        
+     // Assign buried PRL to this hash
+     hashBalances[hash] += claimAmount;
+        
+     // Assign broker to this hash
+     buryBroker[hash] = msg.sender;
+        
+     // Set buried state to true
+     buried[hash] = true;
```

Comment:
It looks like you have removed the claim clock stuff, unless I'm reading everything.  What is in place to force treasure claiming to be spaced out over different epochs?

Reply:
I bumbed into another problem. As we know as soon as a hash is claimed once the privateKey is publicly viewable on the blockchain. This means that for multiepoch uploads not only the webnodes but anyone with access to the ethereum blockchain can keep track of privateKeys to claim them as soon as an epoch is over. We could might be able to check if the privateKey matches on clientside so it does not get exposed but I thing the better solution would be to either bury multiple privateKeys or that the brokers call the bury function multiple time - keep in mind that the bury function is fairly expensive to call though.